### PR TITLE
refactor: convert TableTotals to typescript

### PR DESCRIPTION
**Pull request by Donya-qlik**

### CODE REVIEW

File: src/table/components/__tests__/TableTotals.spec.tsx [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+  let layout: TableLayout;
+  let selectionsAPI: stardust.ObjectSelections;
+  let keyboard: stardust.Keyboard;
 
-  const renderTableTotals = (cellCoordMock) =>
+  const renderTableTotals = (cellCoordMock?: [number, number]) =>
     render(
       <TableContextProvider selectionsAPI={selectionsAPI} cellCoordMock={cellCoordMock}>
-        {getTotalPosition(layout) === 'top' && (
+        {((getTotalPosition(layout) === 'top') as unknown as HTMLElement) && (
```
Comment (Donya-qlik) [inline src/table/components/__tests__/TableTotals.spec.tsx:23]: It's not a good idea to set the string type as HTMLElement but I didn't have another option in my mind to fix the type conflict here... the total position check here makes the problem

**Comment by LiKang6688** [comment]
```
TableTotals.propTypes = {
  rootElement: PropTypes.object.isRequired,
  tableData: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
  layout: PropTypes.object.isRequired,
  keyboard: PropTypes.object.isRequired,
};
```
should be removed

File: src/table/components/__tests__/TableTotals.spec.tsx (hunk 1 above)
Comment (LiKang6688) [reply src/table/components/__tests__/TableTotals.spec.tsx:23]: How about
```
{getTotalPosition(layout) === 'top' ? (
          <TableTotals
            rootElement={rootElement}
            tableData={tableData}
            theme={theme}
            layout={layout}
            keyboard={keyboard}
          />
        ) : (
          <></>
        )}
```

File: src/table/components/__tests__/TableTotals.spec.tsx [hunk 2]
```diff
@@ -39,28 +39,27 @@ describe('<TableTotals />', () => {
...
+    } as stardust.ObjectSelections;
   });
 
   it('should show the total row when table total is rendered', () => {
     layout.totals.position = 'top';
     const { queryByText } = renderTableTotals();
-    expect(queryByText(tableData.columns[0].totalInfo)).toHaveTextContent('Totals');
-    expect(queryByText(tableData.columns[1].totalInfo)).toHaveTextContent('350');
+    expect(queryByText(tableData.columns[0].totalInfo as Matcher)).toHaveTextContent('Totals');
+    expect(queryByText(tableData.columns[1].totalInfo as Matcher)).toHaveTextContent('350');
```
Comment (LiKang6688) [inline src/table/components/__tests__/TableTotals.spec.tsx:62]:
```suggestion
    expect(queryByText(tableData.columns[0].totalInfo as string)).toHaveTextContent('Totals');
    expect(queryByText(tableData.columns[1].totalInfo as string)).toHaveTextContent('350');
```

File: src/table/components/__tests__/TableTotals.spec.tsx [hunk 3]
```diff
@@ -71,14 +70,14 @@ describe('<TableTotals />', () => {
     it('should call handleTotalKeyDown when keyDown on a total cell', () => {
       jest.spyOn(handleKeyPress, 'handleTotalKeyDown').mockImplementation(() => jest.fn());
       const { queryByText } = renderTableTotals();
-      fireEvent.keyDown(queryByText(tableData.columns[0].totalInfo));
+      fireEvent.keyDown(queryByText(tableData.columns[0].totalInfo as Matcher) as Element);
```
Comment (LiKang6688) [inline src/table/components/__tests__/TableTotals.spec.tsx:73]: same as above

File: src/table/components/__tests__/TableTotals.spec.tsx [hunk 4]
```diff
@@ -71,14 +70,14 @@ describe('<TableTotals />', () => {
...
       expect(handleKeyPress.handleTotalKeyDown).toHaveBeenCalledTimes(1);
     });
 
     it('should call removeAndFocus when clicking a total cell', () => {
       jest.spyOn(handleAccessibility, 'removeAndFocus').mockImplementation(() => jest.fn());
       const { queryByText } = renderTableTotals();
-      fireEvent.mouseDown(queryByText(tableData.columns[0].totalInfo));
+      fireEvent.mouseDown(queryByText(tableData.columns[0].totalInfo as Matcher) as Element);
```
Comment (LiKang6688) [inline src/table/components/__tests__/TableTotals.spec.tsx:80]: same as above

File: src/types.ts [hunk 5]
```diff
@@ -314,3 +314,10 @@ export interface RootProps {
   footerContainer: HTMLElement;
   announce: Announce;
 }
+export interface TableTotalsProps {
```
Comment (LiKang6688) [inline src/types.ts:317]: add a new line for the interface, please.

File: src/table/components/__tests__/TableTotals.spec.tsx (hunk 1 above)
Comment (Donya-qlik) [reply src/table/components/__tests__/TableTotals.spec.tsx:23]: yes it makes sense and works properly, thanks!

File: src/table/components/TableTotals.tsx [hunk 6]
```diff
@@ -16,7 +17,7 @@ function TableTotals({ rootElement, tableData, theme, layout, keyboard }) {
   return (
     <StyledHeadRow paginationNeeded={paginationNeeded} className="sn-table-row">
       {columns.map((column, columnIndex) => {
-        const cellCoord = [isTop ? 1 : rows.length + 1, columnIndex];
+        const cellCoord: [number, number] = [isTop || !rows ? 1 : rows.length + 1, columnIndex];
```
Comment (Donya-qlik) [inline src/table/components/TableTotals.tsx:20]: I added more statement on rows here but I don't think I should do, it doesn't make sense to me that rows could be undefined but apparently we have a case for it in TableData interface

Comment (LiKang6688) [reply src/table/components/TableTotals.tsx:20]: Try to merge the main branch with your branch.

File: src/table/components/TableTotals.tsx [hunk 7]
```diff
@@ -5,8 +5,9 @@ import { getTotalsCellStyle } from '../utils/styling-utils';
 import { handleTotalKeyDown } from '../utils/handle-key-press';
 import { removeAndFocus } from '../utils/handle-accessibility';
 import { StyledHeadRow, StyledTotalsCell } from '../styles';
+import { TableTotalsProps } from '../../types';
```
Comment (LiKang6688) [inline src/table/components/TableTotals.tsx:8]: import PropTypes from 'prop-types'; in line 2 can be removed

File: src/table/components/TableTotals.tsx [hunk 8]
```diff
@@ -16,7 +17,7 @@ function TableTotals({ rootElement, tableData, theme, layout, keyboard }) {
   return (
     <StyledHeadRow paginationNeeded={paginationNeeded} className="sn-table-row">
       {columns.map((column, columnIndex) => {
-        const cellCoord = [isTop ? 1 : rows.length + 1, columnIndex];
+        const cellCoord: [number, number] = [isTop ? 1 : rows.length + 1, columnIndex];
```
Comment (LiKang6688) [inline src/table/components/TableTotals.tsx:20]:
```suggestion
        const cellCoord = [isTop ? 1 : rows.length + 1, columnIndex];
```

File: src/table/components/__tests__/TableTotals.spec.tsx [hunk 9]
```diff
@@ -1,33 +1,35 @@
 import React from 'react';
-import { render, fireEvent } from '@testing-library/react';
+import { render, fireEvent, Matcher } from '@testing-library/react';
```
Comment (LiKang6688) [inline src/table/components/__tests__/TableTotals.spec.tsx:2]: Matcher can be removed

**Review by LiKang6688 — APPROVED** [review]
Two unused ones can be removed. Otherwise looks good.

**Pull request merged**